samples/bluetooth/bap_broadcast_sink/src/main.c

```diff
@@ -281,83 +289,115 @@ static bool subgroup_get_valid_bis_indexes_cb(const struct bt_bap_base_subgroup
+			       bis_indexes);
 		}
 	}
+
+	bis_index->num_subgroups++;
+
 	return true; /* continue to next subgroup */
 }
+#else /* !CONFIG_TARGET_BROADCAST_CHANNEL */
+static bool subgroup_get_bis_indexes_cb(const struct bt_bap_base_subgroup *subgroup,
+					void *user_data)
+{
+	struct bis_index_data *bis_index = user_data;
+	const int err = bt_bap_base_subgroup_get_bis_indexes(
+		subgroup, &bis_index->bis_index_bitfield[bis_index->num_subgroups]);
+
+	if (err != 0) {
+		printk("Failed to parse all BIS: %d", err);
+
+		return false;
+	}
+
+	printk("bis_index.bis_index_bitfield[%d] = 0x%0x\n", bis_index->num_subgroups,
+	       bis_index->bis_index_bitfield[bis_index->num_subgroups]);
+
+	bis_index->num_subgroups++;
+
+	return true;
+}
 #endif /* CONFIG_TARGET_BROADCAST_CHANNEL */
 
 static void base_recv_cb(struct bt_bap_broadcast_sink *sink, const struct bt_bap_base *base,
 			 size_t base_size)
 {
-	uint32_t base_bis_index_bitfield = 0U;
 	int err;
 
+	base_bis_index_bitfield = 0U;
+
 	if (base_received) {
 		return;
 	}
 
 	printk("Received BASE with %d subgroups from broadcast sink %p\n",
 	       bt_bap_base_get_subgroup_count(base), sink);
 
+	memset(&bis_index, 0, sizeof(struct bis_index_data));
+
 #if defined(CONFIG_TARGET_BROADCAST_CHANNEL)
 	/**
 	 * Get a 32-bit bitfield of BIS indexes that cover the channel allocation values in
 	 * CONFIG_TARGET_BROADCAST_CHANNEL.
 	 */
 	printk("Target channel location: 0x%x\n", CONFIG_TARGET_BROADCAST_CHANNEL);
-	err = bt_bap_base_foreach_subgroup(base, subgroup_get_valid_bis_indexes_cb,
-					   &base_bis_index_bitfield);
-	if ((err != 0 && err != -ECANCELED) ||
-	    (err == -ECANCELED && base_bis_index_bitfield == 0)) {
-		printk("Failed to get valid BIS indexes: %d\n", err);
-
-		return;
-	}
+	err = bt_bap_base_foreach_subgroup(base, subgroup_get_valid_bis_indexes_cb, &bis_index);
 #else
-	err = bt_bap_base_get_bis_indexes(base, &base_bis_index_bitfield);
-	if (err != 0) {
-		printk("Failed to get BIS indexes: %d\n", err);
+	/* Get BIS index data for each subgroup */
+	err = bt_bap_base_foreach_subgroup(base, subgroup_get_bis_indexes_cb, &bis_index);
+#endif /* CONFIG_TARGET_BROADCAST_CHANNEL */
 
-		return;
+	for (uint8_t subgroup = 0; subgroup < bis_index.num_subgroups; subgroup++) {
+		base_bis_index_bitfield |= bis_index.bis_index_bitfield[subgroup];
 	}
 
-#endif /* CONFIG_TARGET_BROADCAST_CHANNEL */
+	if (err != 0 || base_bis_index_bitfield == 0) {
+		printk("Failed to get valid BIS indexes: %d\n", err);
 
-	bis_index_bitfield = base_bis_index_bitfield & bis_index_mask;
+		return;
+	}
 
-	printk("bis_index_bitfield = 0x%08x\n", bis_index_bitfield);
+	printk("base_bis_index_bitfield = 0x%08x\n", base_bis_index_bitfield);
 
 	if (broadcast_assistant_conn == NULL) {
 		/* No broadcast assistant requesting anything */
 		requested_bis_sync = BT_BAP_BIS_SYNC_NO_PREF;
+		requested_subgroup_sync = (1U << CONFIG_BT_BAP_BASS_MAX_SUBGROUPS) - 1;
```

Comment:
```suggestion
		requested_subgroup_sync = BIT_MASK(CONFIG_BT_BAP_BASS_MAX_SUBGROUPS);
```

I assume the purposes here is to the set lowest CONFIG_BT_BAP_BASS_MAX_SUBGROUPS bits to 1?

Reply:
Please also add a comment stating this, as that is (IMO) non-obvious